### ISSUE TITLE
null exception from filter with Query-Time Lookup

### ISSUE THREAD

**Issue by epistemancer** [apache/incubator-druid#1711]
If a query contains a filter with a QTL extractionFn, and its intervals contain both times for which there is data and times for which there is no data, Druid returns a null exception.

For example, it's currently about 2015-09-09 00:15 UTC, and it looks like Druid contains data through about 2015-09-08 23:30.  A query like:

```
{
  "queryType":"timeseries",
  "dataSource":"metrics",
  "granularity":"all",
  "intervals":["2015-09-02/2015-09-09"],
  "filter":{
    "type":"extraction",
    "dimension":"service",
    "value":"hadoop",
    "extractionFn":{
      "type":"regex",
      "expr":"^([^/]*).*"
    }
  },
  "aggregations":[
    {"name":"size","type":"cardinality","fieldNames":["service"],"byRow":true}
  ]
}
```

returns
`{"error":"null exception"}`

But the same query over a period with
`"intervals":["2015-09-02/2015-09-08T23:00:00"]`
succeeds.

Similarly, if the time range is changed to one for which the whole interval contains no data, e.g. one entirely in the future or entirely in the past before the first datum in Druid:
`"intervals":["2010-09-02/2010-09-09"]`
then Druid successfully returns `[]`.

It's apparently only when the intervals contain overlaps both periods for which there is data and periods for which there is no data that there's a problem.

**Comment by epistemancer** [comment]
Looks like this may have already been resolved by:
https://github.com/druid-io/druid/pull/1578

**Comment by xvrl** [comment]
@epistemancer there was also a bug with regex extraction on null values #1731

**Comment by vogievetsky** [comment]
Jonathan! Long time no see! I remember when this was filed... good times.
@fjy I believe this was resolved (as noted above) by #1578 and #1731 and can be closed.
